vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -163,6 +168,36 @@ void inject_slot_features(example* shared, example* slot)
   }
 }
 
+void inject_slot_id(ccb& data, example* shared, int id)
+{
+  // id is zero based, so the vector must be of size id + 1
+  if(id + 1 > data.hashes.size())
```

Comment:
This is minor, but the code is cleaner if you remove 'if', change the next 'if' to do the array bounds check, and then use push_back() inside the next 'if'.

Reply:
Yes we could do that with the assumption that id is always increasing one at a time